instrumentation/trilogy/test/opentelemetry/instrumentation/trilogy/instrumentation_test.rb

```diff
@@ -346,6 +346,46 @@
       end
     end
 
+    describe 'when propagator is set to none' do
+      let(:config) { { propagator: :none } }
+
+      it 'does not inject context' do
+        sql = +'SELECT * from users where users.id = 1 and users.email = "test@test.com"'
+        original_sql = sql.dup
+        expect do
+          client.query(sql)
+        end.must_raise Trilogy::Error
+        _(sql).must_equal original_sql
+      end
+    end
+
+    describe 'when propagator is set to nil' do
+      let(:config) { { propagator: nil } }
+
+      it 'does not inject context' do
+        sql = +'SELECT * from users where users.id = 1 and users.email = "test@test.com"'
+        original_sql = sql.dup
+        expect do
+          client.query(sql)
+        end.must_raise Trilogy::Error
+        _(sql).must_equal original_sql
+      end
+    end
+
+    describe 'when propagator is set to vitess' do
+      let(:config) { { propagator: 'vitess' } }
+
+      it 'does inject context' do
+        sql = +'SELECT * from users where users.id = 1 and users.email = "test@test.com"'
+        original_sql = sql.dup
+        expect do
+          client.query(sql)
+        end.must_raise Trilogy::Error
+        encoded = Base64.strict_encode64("{\"uber-trace-id\":\"#{span.hex_trace_id}:#{span.hex_span_id}:0:1\"}")
+        _(sql).must_equal "/*VT_SPAN_CONTEXT=#{encoded}*/#{original_sql}"
+      end
+    end
+
```

Comment:
Will there be conflicts or issues with comments generated by ActiveRecord `annotate` or `query_log_tags` when the user anticipates those tags to be prepended? 

https://api.rubyonrails.org/classes/ActiveRecord/QueryLogs.html

Reply:
Unclear as yet. We're going to try this out in production and see.

Reply:
Keen on hearing the results